go/stream/message.go

```diff
@@ -49,6 +51,9 @@ type MessageProperties struct {
 	WebhookFailureReason string    `json:"webhookFailureReason,omitempty"` // optional
 	Stage                string    `json:"stage,omitempty"`                // optional
 	Compression          string    `json:"compression,omitempty"`          // optional
+	Encryption           string    `json:"encryption,omitempty"`           // optional
+	// if key is rotated EncryptionKeyID should be used to refer to correct key
+	EncryptionKeyID string `json:"encryptionKeyID,omitempty"` // optional
```

Comment:
@mihir20 Why not use a single message property for this? We can store the algorithm and any potential settings after a separator (e.g. a semicolon).

For example:
```
aes256;/path/to/key55
```

When a secret is rotated the old one will also be kept. So following the example above, in the disk you will also have `/path/to/key54`.

If the file isn't there then you just have to raise an error and say you cannot encrypt/decrypt the message due to the missing key.

It will be up to the algorithm rotating the keys to make sure you'll always have at least the last and second last keys on disk.

Reply:
It doesn't have to be a file necessarily btw, it can also be a key like you're doing and then you check an env var with the same name. My point being, can we have both in the same message property? Wdyt?

Reply:
Yeah, I thought about keeping both the settings and ID in same field but that would become messy. Current approach feels more cleaner. 
Is there any constraint on number of fields we can keep in properties?

Reply:
> Is there any constraint on number of fields we can keep in properties?

Can't you check? 

Anyway, I'm not proposing this due to a hard limit but because it seems easy to achieve and not messy at all given that you are already storing the algorithm and its settings with a separator so you just need another separator and add the key ID. I'm also suggesting this because we already have many properties so if each time we don't want to add a separator we add a property instead then it's going to become "messy". Also, we don't necessarily have to merge them but I think it'd be nice for the above reasons.

Reply:
@mihir20 please poke other team members about this to get their opinion.

Reply:
> Is there any constraint on number of fields we can keep in properties?

I did checked this, but nothing much is mentioned in the Docs regarding `properties`.